server/src/main/scala/org/elasticmq/server/config/ElasticMQServerConfig.scala

```diff
@@ -68,47 +71,75 @@ class ElasticMQServerConfig(config: Config) extends Logging {
 
   val restStatisticsConfiguration = new RestStatisticsConfiguration
 
+  private val queuesStorage = config.getConfig("queues-storage")
+  val queuesStoragePath: String = queuesStorage.getString("path")
+  val queuesStorageEnabled: Boolean = queuesStorage.getBoolean("enabled")
+
+  def readQueuesToLoad(persistedQueues: Seq[CreateQueue]): Seq[CreateQueue] = {
+    val persistedQueuesName = persistedQueues.map(_.name)
+    val baseQueues = createBaseQueues.filterNot(queue => persistedQueuesName.contains(queue.name))
+    persistedQueues ++ baseQueues
+  }
+
+  val createBaseQueues: Seq[CreateQueue] = {
+    val baseQueues: mutable.Map[String, ConfigValue] = config
+      .getObject("queues")
+      .asScala
+    createQueuesFromConfig(baseQueues)
+  }
+
+  private val persistedQueuesConfig: Option[Config] =
+    if (queuesStorageEnabled)
+      Some(
+        ConfigFactory
+          .parseFile(new File(queuesStoragePath))
+      )
+    else None
+
+  def createPersistedQueues(backupFile: Option[Config] = persistedQueuesConfig): List[CreateQueue] =
```

Comment:
the flow here isn't easy to understand. First, maybe let's switch from `createPersistenQueues` to `readPersistedQueues` (and same above). I know the name used to be `createQueues` but that was just a copy of the result type, probably not the most readable name as well :)

Secondly, the main entry point here is `readQueuesToLoad` which in production will always be called with the `createPersistedQueues` parameter - so maybe let's make it the default?